deployer/handlers.go

```diff
@@ -182,7 +187,10 @@ func CleanUpSuccess(awsc aws.Clients) DeployHandler {
 			return nil, &errors.CleanUpError{err.Error()}
 		}
 
-		if err := release.UnlockRoot(awsc.S3Client(release.AwsRegion, nil, nil)); err != nil {
+		locker := dynamodb.NewDynamoDBLocker(awsc.DynamoDBClient(nil, nil, nil))
+		lockTableName := getLockTableNameFromContext(ctx, "-locks")
+
+		if err := release.UnlockRoot(awsc.S3Client(release.AwsRegion, nil, nil), locker, lockTableName); err != nil {
```

Comment:
Not sure if Bob's "go style" comment about this fundamentally being a statement to "UnlockRoot", rather than a conditional applies here, but it kinda seems like it does.

Reply:
To be honest, though I agree with the comment from a general / non-Go point-of-view, I don't think it's necessarily a comment that is held strongly by most of the Go community. 

Indeed, when in doubt, I tend to look towards the stdlib, and the official blog posts. And you can actually find evidence to suggest the contrary. 

e.g. https://blog.golang.org/error-handling-and-go

There is also a decent blog post on this issue: https://www.calhoun.io/one-liner-if-statements-with-errors/